MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -2,71 +2,172 @@ import UIKit
+    
+    @IBOutlet private weak var counterLabel: UILabel!
+    
+    @IBOutlet private weak var textLabel: UILabel!
+    
+    private var correctAnswers:Int = 0
+    
+    private var currentQuestionIndex: Int = 0
+    var currentQuestion: QuizQuestion {
+        get{
+            questions[currentQuestionIndex]
+        }
+    }
+    
     override func viewDidLoad() {
         super.viewDidLoad()
+        let viewModel = convert(model: currentQuestion)
+        show(quiz: viewModel)
+    }
+    
+    private func checkAnswer (answer: Bool) -> Bool {
+        if currentQuestion.correctAnswer == answer {
+            correctAnswers += 1
+        }
+        return currentQuestion.correctAnswer == answer
+    }
+    
+    @IBAction func NoButtonDidTap(_ sender: Any) {
+        let isCorrect = checkAnswer(answer: false)
+        showAnswerResult(isCorrect: isCorrect)
+    }
+    
+    @IBAction func YesButtonDidTap(_ sender: Any) {
+        let isCorrect = checkAnswer(answer: true)
+        showAnswerResult(isCorrect: isCorrect)
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 6
+
+        if isCorrect == true {
+            imageView.layer.borderColor = UIColor.ypGreen.cgColor
+        } else {
+            imageView.layer.borderColor = UIColor.ypRed.cgColor
+        }
+        // сюда бы я добавила дизейбл кнопки, но мне лень
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { self.showNextQuestionOrResults()
+            self.imageView.layer.borderWidth = 0
+        }
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+      // здесь мы заполняем нашу картинку, текст и счётчик данными
+        imageView.image = step.image
+        counterLabel.text = step.questionNumber
+        textLabel.text = step.question
     }
-}
-
-/*
- Mock-данные
- 
- 
- Картинка: The Godfather
- Настоящий рейтинг: 9,2
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
-
-
- Картинка: The Dark Knight
- Настоящий рейтинг: 9
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
-
-
- Картинка: Kill Bill
- Настоящий рейтинг: 8,1
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
-
-
- Картинка: The Avengers
- Настоящий рейтинг: 8
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
-
-
- Картинка: Deadpool
- Настоящий рейтинг: 8
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
-
-
- Картинка: The Green Knight
- Настоящий рейтинг: 6,6
- Вопрос: Рейтинг этого фильма больше чем 6?
- Ответ: ДА
 
+    private func show(quiz result: QuizResultsViewModel) {
+      // здесь мы показываем результат прохождения квиза
+        let alert = UIAlertController (title: result.title, message: result.text, preferredStyle: .alert)
+        
+        let action = UIAlertAction (title: result.buttonText, style: .default) { _ in self.currentQuestionIndex = 0
```

Comment:
1) После ключевого слова перенос строки 
2) в будущих спринтах будете изучать память и слабые ссылки, пока предлагаю сделать так:
   1) в квадратных скобках применить "ослабление ссылки"
   2) конструкцией guard разыменовать слабую ссылку на self 
Опять же, уберегаем себя от ошибок в будущем 
```suggestion
        let action = UIAlertAction (title: result.buttonText, style: .default) { [weak self] _ in 
            guard let self = self else { return }
            self.currentQuestionIndex = 0
```